File: Exams/E3-RestAPI/JPA/src/main/java/com/danielmunoz/psp/jpa/data/Artist.java
package com.danielmunoz.psp.jpa.data;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Artist {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private String name;

    @Column
    private LocalDate birthDate;

    @Column
    private String city;

    @Column
    private Integer monthlyListeners;

    @Column
    private String label;

    @ManyToMany(mappedBy = "artists")
    @JsonIgnore
    private List<Album> albums;

}

File: Exams/E3-RestAPI/src/main/kotlin/com/raks/jpa/entity/Artist.kt
package com.raks.jpa.entity

import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "artists")
open class Artist {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    open var id:               Int?               = null

    @Column(name = "name", nullable = false)
    open var name:             String?            = null

    @Column(name = "birth_date", nullable = false)
    open var birthDate:        LocalDate?         = null

    @Column(name = "city", nullable = false)
    open var city:             String?            = null

    @Column(name = "monthly_listeners")
    open var monthlyListeners: Int?               = null

    @Column(name = "label", nullable = false)
    open var label:            String?            = null

    @ManyToMany(mappedBy = "artists")
    open var albums:           MutableList<Album> = mutableListOf()

}